### DISCUSSION TITLE
Programmatically Exit a WebXR session?

### DISCUSSION THREAD

**Discussion by milesgreen** [google/model-viewer#2133] [Q&A] [answered]
Is it possible to programmatically exit the WebXR session, beyond using the exit-webxr-ar-button slot?

For example, if additional interactive DOM elements have been placed on top of the WebXR view and interacting with those elements should update and reveal content on the page running behind the WebXR session.

**Comment by elalish** [comment, marked as answer]
I'm not sure how your example relates to your question. The rest of your page still exists and can be modified, it's just that only the parts that are children of the model-viewer element are visible in WebXR mode. For programmatically exiting, yes you'll need to dispatch a click to the exit button slot. If that's difficult, let us know. It would not be hard to expose that as a method.

**Reply by milesgreen** [reply]
Cool. Perfect. I'll look at dispatching a click to the exit button slot.

Regrading my example, to be more specific, let's say I want to show an 'Add To Cart' button on top of the WebXR session (which I can do with extra DOM elements within the `<model-viewer>` element), and when the user clicks that button I want to update a shopping cart via AJAX (which I can do with regular JS on the page that's still behind the WebXR session) BUT I want to exit the WebXR session so that the user can see the item has been added to their cart (which I currently couldn't do). So I'll give dispatching a click a go. 

Thanks

**Reply by milesgreen** [reply]
I managed to get this to work with something along the lines of:

`modelViewer.shadowRoot.querySelector('#default-exit-webxr-ar-button').click();`

And that totally enabled me to close the WebXR session. Thanks.
